src/__tests__/index.test.ts

```diff
@@ -0,0 +1,512 @@
+import util from 'util';
+import assert from 'assert';
+import { stub, SinonStub } from 'sinon';
+import oracledb, { Connection, PoolAttributes } from 'oracledb';
+import OracleWrapper from '../index';
+import { Logger, OracleDBOptions } from '../types';
+
+describe('OracleWrapper', function () {
+  let wrapper: OracleWrapper;
+  let getConnectionStub: SinonStub;
+  let createPoolStub: SinonStub;
+  let connectionPool: Record<string, SinonStub>;
+  let dbConnection: Record<string, SinonStub>;
+  let queryResult: Record<string, Record<string, SinonStub>>;
+  const credentials = {
+    host: '127.0.0.1',
+    port: '1521',
+    sid: 'ORCTEST',
+    user: 'SYSADMIN',
+    password: 'oracledbtest',
+  };
+  let options: OracleDBOptions;
+  let testLogger: Record<string, SinonStub>;
+
+  beforeEach(function () {
+    testLogger = {
+      log: stub(),
+      error: stub(),
+      warn: stub(),
+      info: stub(),
+      debug: stub(),
+      verbose: stub(),
+    };
+    connectionPool = {
+      close: stub(),
+    };
+    dbConnection = {
+      execute: stub(),
+      release: stub(),
+    };
+    queryResult = {
+      resultSet: {
+        getRows: stub(),
+        close: stub(),
+      },
+    };
+    options = {
+      alias: 'TestDB1',
+      logger: testLogger as unknown as Logger,
+    };
+    wrapper = new OracleWrapper(credentials, options);
+    getConnectionStub = stub(oracledb, 'getConnection');
+    createPoolStub = stub(oracledb, 'createPool');
+  });
+  describe('Constructor', function () {
+    describe('Alias', function () {
+      const testAlias = 'TestDB2';
+      let aliasedWrapper: OracleWrapper;
+      context('With a value in options', function () {
+        beforeEach(function () {
+          aliasedWrapper = new OracleWrapper(credentials, { alias: testAlias });
+        });
+        it('Should use the alias as the property', function () {
+          // eslint-disable-next-line
+          assert.strictEqual(aliasedWrapper['alias'], testAlias);
+        });
+        it('Should use the alias in the poolCredentials', function () {
+          // eslint-disable-next-line
+          assert.strictEqual(aliasedWrapper['poolCredentials'].poolAlias, testAlias);
```

Comment:
There are a few instances of `eslint-disable-next-line`. Might it be better to either use something more targeted like `// eslint-disable-next-line @typescript-eslint/dot-notation` or update the ruleset to allow for private property access like this?

Reply:
Good point. I narrowed that down in eb4b8de. I'd rather not tweak the ruleset for this; it's really just a workaround to access the private properties and I think it's something we'd want to be deliberate about.